CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+Welcome to Stream’s Android repository! We welcome all feedback and we hope this document helps you contribute to our repo! 🎉.
+
+We'll guide you through our repository structure, present our styling guidelines and describe the pull request and approval process.
+
+These are guidelines and don't have to be followed to a letter, if you're looking to contribute, feel free to do your best and what you're comfortable with, and we'll take care of the rest! 🤗
+
+---
+
+# If I have a question, do I need to read this guide? 💬
+
+Not really! This guide is focused on contributions, so if you have questions, want to report a bug or want to request a new feature, use our [Android Chat SDK Task and Bug Report](https://github.com/GetStream/stream-chat-android/issues/new/choose) templates.
+
+We'll do our best to respond as soon as possible and provide you with guidance and the more details you can provide, the easier it is for us to understand what you need help with!
+
+---
+
+# What should I know before diving into code? 🤔
+
+Our Android repository contains several modules that each represent an artifact on Maven, as described in our [documentation](https://getstream.io/chat/docs/sdk/android/basics/dependencies/). We also have a few modules related to documentation and testing, which are not published as libraries.
+
+If you haven't already, make sure to read our [README](https://github.com/GetStream/stream-chat-android/blob/main/README.md), to learn all about the repo, our codebase, documentation and much more.
+
+![Android Repository Structure](https://user-images.githubusercontent.com/17215808/173359688-31ec7680-0a59-4ad6-91a4-f538a935ce26.PNG)
+
+### Important modules & files 🧱
+
+- `stream-chat-android-client`: Holds all the code connected to the **low-level-client**, used to request data from and send information to the API.
+- `stream-chat-android-offline`: As an addition to the **client** module, you can use **offline** to introduce persistence and caching to your app.
+- `stream-chat-android-ui-common`: Common UI-related code and utility functions.
+
+- `stream-chat-android-compose-sample`: The main sample app featuring the Compose SDK. 
+- `stream-chat-android-compose`: Our Compose SDK that features channels and messages components, as well as other utilities.
+- `stream-chat-android-ui-components-sample`: The main sample app featuring the XML (UI Components) SDK. 
+- `stream-chat-android-ui-components`: Our XML (UI Components) SDK that features channels and messages components, as well as other utilities.
```

Comment:
Let me make a modularization graph like this when I have a moment. 😄  https://github.com/GetStream/AvengersChat/blob/main/figure/figure4.png

Reply:
Nice! We should also include `stream-chat-android-state`

Reply:
@skydoves did you perhaps have any time for the diagram? If not, we could just merge this for now and then add the diagram at some point in the future

Reply:
Let's merge this PR first. I will make a new PR for the diagram in the near future.